pkg/stores/partition/listprocessor/processor.go

```diff
@@ -258,19 +258,16 @@ func getLimit(apiOp *types.APIRequest) int {
 	return limit
 }
 
-// FilterList accepts a channel of unstructured objects and a slice of filters and returns the filtered list.
+// FilterList accepts a slice of unstructured objects and a slice of filters and returns the filtered list.
 // Filters are ANDed together.
-func FilterList(list <-chan []unstructured.Unstructured, filters []OrFilter) []unstructured.Unstructured {
+func FilterList(list []unstructured.Unstructured, filters []OrFilter) []unstructured.Unstructured {
+	if len(filters) == 0 {
+		return list
+	}
```

Comment:
Before I think we had a difference in return vs input type, so that wasn't really possible there. Ordinarily I would fix a small inconsistency like this to avoid difficult-to-locate bugs (like someone changing the original list in a go-routine and not realizing that it affected other values down-the-line), but I think there's a performance impact here to consider - copying the items to another list would have a non-zero performance impact. I would also expect the "no filters" case to be the default for the time being, so I'd like to keep this as-is and avoid a costly copy operation.

WDYT?

Reply:
I'm okay as is. I think consistency would be better from my POV. The costly copy was already there, so adding it back wouldn't make rancher perform worse than it already is. We could benchmark it and see how much optimizing this improves the performance in the grand scheme of things but I'd rather do that as a separate issue and as a need basis.

Reply:
Without benchmark, it's usually difficult to know if an optimization improves or not. For example, the compiler could already be optimizing this part of the code. It's hard to tell without benchmarks.